test.js

```diff
@@ -1,45 +1,69 @@
-const test = require('tape').test
+const test = require('node:test')
 const persistence = require('./')
 const Redis = require('ioredis')
 const mqemitterRedis = require('mqemitter-redis')
 const abs = require('aedes-cached-persistence/abstract')
-const db = new Redis()
 
-db.on('error', e => {
-  console.trace(e)
-})
+function sleep (sec) {
+  return new Promise(resolve => setTimeout(resolve, sec * 1000))
+}
 
-db.on('connect', unref)
+function waitForEvent (obj, resolveEvt) {
+  return new Promise((resolve, reject) => {
+    obj.on(resolveEvt, () => {
+      resolve()
+    })
+    obj.on('error', reject)
```

Comment:
maybe it's better to use `once` here?

Reply:
Both work, but .once is more clean indeed :-)